### DISCUSSION TITLE
Disabling sleep mode for some rigid bodies

### DISCUSSION THREAD

**Discussion by frutas-fruit** [xissburg/edyn#95] [General]

**Comment by xissburg** [comment]
Are you refreshing or marking the car entity as dirty when you apply these inputs?

**Reply by xissburg** [reply]
Continuous means that it will be updated into the main thread after every step in the island worker, not the other way around. You still need to notify changes in the main thread to make sure they're pushed to the island worker (which should wake it up as well if it's sleeping).

Either way, you can set `sleeping_disabled` to true in the `rigidbody_def` to prevent sleeping.

**Reply by xissburg** [reply]
The reason the car stays in the air is probably because the changes you're making in the main thread are being overwritten by the messages coming from the island worker in every update, since it's marked as continuous. 

I am not sure what is your setup with EdynCar, but maybe you want to split things up and avoid having one component that does everything. You could have input components which are only modified in the main registry and marked as dirty to update them into the workers and other component containing vehicle status which can be marked as continuous to bring updated info from the background threads back into the main registry.

**Reply by xissburg** [reply]
You shouldn't tag `InputStruct` as continuous. You'll have to mark it as dirty when it changes. The continuous tag only works the other way around, to always update it from background thread into main thread. It is used primarily for transforms and velocities since these change after every step so the island worker in the background thread will always include it in the set of updated components and make sure it stays up to date in the main registry. What's in the `EdynCar` component?

**Reply by xissburg** [reply]
Yes, that's how you mark dirty. The `EdynCar` struct is something you're writing to in the external system and reading from in the main thread so it would make sense to mark it as continuous.

**Comment by frutas-fruit** [comment]


**Reply by xissburg** [reply]
When the rigid body goes to sleep, the island stops execution so the external system won't be called.
